pkg/filters/proxies/grpcproxy/pool.go

```diff
@@ -87,7 +87,7 @@ func (m *MultiPool) Get(ctx stdcontext.Context) (objectpool.PoolObject, error) {
 	if value, ok = m.pools.Load(key); !ok {
 		m.lock.Lock()
 		defer m.lock.Unlock()
-		if value, _ = m.pools.Load(key); !ok {
+		if value, ok = m.pools.Load(key); !ok {
 			value = objectpool.NewWithSpec(m.spec, ctx)
 			defer m.pools.Store(key, value)
 		}
```

Comment:
don't use `defer` here, as the defered function will be called at the exit of the current function.

```go
if value, ok = m.pools.Load(key); !ok {
		m.lock.Lock()
		if value, _ = m.pools.Load(key); !ok {
			value = objectpool.NewWithSpec(m.spec, ctx)
			m.pools.Store(key, value)
		}
		m.lock.Unlock()
}
```